src/components/upload/progress-card-combined.tsx

```diff
@@ -1,34 +1,27 @@
-import type { Progress } from '../../types/upload-progress.ts'
-import { getEstimatedTime, getStepLabel } from '../../utils/upload-status.ts'
+import type { StepState } from '../../types/upload/step.ts'
+import { getStepEstimatedTime, getStepLabel } from '../../utils/upload/step.ts'
 import { Card } from '../ui/card.tsx'
 import { ProgressBar } from '../ui/progress-bar.tsx'
 
 interface ProgressCardCombinedProps {
-  progresses: Array<Progress>
-  getCombinedFirstStageStatus: () => Progress['status']
-  getCombinedFirstStageProgress: () => number
+  stepStates: Array<StepState>
+  firstStageStatus: StepState['status']
+  firstStageProgress: StepState['progress']
```

Comment:
I know we already had this code (and I think I added it), but I have not been a fan of explicitly declaring the "firstStage" as it's own prop.. ideally this would be codified in the "StepState" data model and handled appropriately, rather than this explicit naming of the first step.

ultimately though, the goal is simplicity since we want folks to be able to follow the code here easily.